drivers/syslog/vsyslog.c

```diff
@@ -132,8 +137,25 @@ int nx_vsyslog(int priority, FAR const IPTR char *fmt, FAR va_list *ap)
 #if defined(CONFIG_SYSLOG_TIMESTAMP)
   /* Pre-pend the message with the current time, if available */
 
+#if defined(CONFIG_SYSLOG_TIMESTAMP_FORMATTED)
+  time = ts.tv_sec;
+#if defined(CONFIG_SYSLOG_TIMESTAMP_LOCALTIME)
+  localtime_r(&time, &tm);
+#else
+  gmtime_r(&time, &tm);
+#endif
+
+  ret = strftime(date_buf, CONFIG_SYSLOG_TIMESTAMP_BUFFER,
+                 CONFIG_SYSLOG_TIMESTAMP_FORMAT, &tm);
+
+  if (ret > 0)
+    {
+      ret = lib_sprintf(&stream.public, "[%s] ", date_buf);
```

Comment:
should we output nanosecond? otherwise, the precision is too low.

Reply:
@xiaoxiang781216 By definition `strftime` is low precision. It cannot accept nanoseconds.  
How do you propose to add nanoseconds?

Reply:
I think that this is an expected compromise. I have "rare" events to log that span between weeks of time, so I prefer to lose precision, to have readability.
If someone else needs precision, and has fast occurring events, then they lose formatting to get the precision. And get increased performance by reducing the logger overhead, needed in "fast" logging systems.

Reply:
like this:
`
ret = lib_sprintf(&stream.public, "[%s.%06ld]] ", date_buf, ts.tv_nsec / 1000);
`

Reply:
I think that this is not correct. It will break the formatting in may cases.  
For example, with `"%H:%M:%S %d %h"` you get:  
`[13:11:27 14 Jan.035468]`, which is at least ugly.

Or  `"%H:%M"`... It will create confusion between seconds and nanoseconds...

I am opposed to making any assumptions on the date format, and altering the formatting that the user selected.

---

If nanosecond accuracy is so important, I would suggest to add another Kconfig option regarding whether nanoseconds shall be appended after the formatted date.  
And possibly have something:  
`[13:11:27 14 Jan] [035468]`
 
What do you think?

Reply:
Ok, let's keep the current accuracy, and watch other user's feedback.